src/azure-cli/azure/cli/command_modules/serviceconnector/_transformers.py

```diff
@@ -57,7 +57,7 @@ def transform_linker_properties(result):
     if is_aks_linker_by_id(resource_id):
         result['kubernetesResourceName'] = get_aks_resource_name(result)
     try:
-        output = run_cli_cmd('az webapp connection list-configuration --id {} -o json'.format(resource_id))
+        output = run_cli_cmd('az webapp connection list-configuration --id "{}" -o json'.format(resource_id))
         result['configurations'] = output.get('configurations')
```

Comment:
Hi @xfz11 , considering potential security risks subprocess might bring, cli recently provided a centralized func `run_cmd` to deal with these issues. 

code is here: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/util.py#L1361-L1372

More guidelines about subprocess security vulnerabilities and fix examples can be found in this doc https://github.com/Azure/azure-cli/blob/282c6e8f4358934094b512e29ae1421438de67aa/doc/cli_subprocess_guidelines.md#cli-centralized-subsystem-executing

Please follow the example and fix related issues.